File: maritime-ai-service/app/api/v1/host_actions.py
"""Host action audit endpoints."""
from fastapi import APIRouter, Request

from app.api.deps import RequireAuth
from app.core.rate_limit import limiter
from app.engine.context.host_action_audit import log_host_action_event
from app.models.schemas import HostActionAuditRequest, HostActionAuditResponse

router = APIRouter(prefix="/host-actions", tags=["host-actions"])


@router.post("/audit", response_model=HostActionAuditResponse)
@limiter.limit("120/minute")
async def submit_host_action_audit(
    request: Request,
    body: HostActionAuditRequest,
    auth: RequireAuth,
) -> HostActionAuditResponse:
    await log_host_action_event(
        event_type=body.event_type,
        user_id=auth.user_id,
        action=body.action,
        request_id=body.request_id,
        summary=body.summary,
        organization_id=auth.organization_id,
        ip_address=request.client.host if request.client else None,
        user_agent=request.headers.get("user-agent"),
        host_type=body.host_type,
        host_name=body.host_name,
        page_type=body.page_type,
        page_title=body.page_title,
        user_role=body.user_role,
        workflow_stage=body.workflow_stage,
        preview_kind=body.preview_kind,
        preview_token=body.preview_token,
        target_type=body.target_type,
        target_id=body.target_id,
        surface=body.surface,
        metadata=body.metadata,
    )
    return HostActionAuditResponse(
        event_type=body.event_type,
        action=body.action,
        request_id=body.request_id,
    )

File: wiii-desktop/src/api/host-actions.ts
import { getClient } from "./client";

export type HostActionAuditEventType =
  | "preview_created"
  | "apply_confirmed"
  | "publish_confirmed";

export interface HostActionAuditRequest {
  event_type: HostActionAuditEventType;
  action: string;
  request_id: string;
  summary?: string;
  host_type?: string;
  host_name?: string;
  page_type?: string;
  page_title?: string;
  user_role?: string;
  workflow_stage?: string;
  preview_kind?: string;
  preview_token?: string;
  target_type?: string;
  target_id?: string;
  surface?: string;
  metadata?: Record<string, unknown>;
}

interface HostActionAuditResponse {
  status: "success";
  event_type: string;
  action: string;
  request_id: string;
}

export async function submitHostActionAudit(
  body: HostActionAuditRequest,
): Promise<HostActionAuditResponse> {
  return getClient().post<HostActionAuditResponse>("/api/v1/host-actions/audit", body);
}
